src/Microsoft.AspNetCore.Mvc.Core/ModelBinding/ParameterBinder.cs

```diff
@@ -297,6 +299,35 @@ public class ParameterBinder
                     modelBindingResult.Model,
                     metadata);
             }
+            else if (metadata.IsRequired)
+            {
+                // We need to special case the model name for cases where a 'fallback' to empty
+                // prefix occurred but binding wasn't successful. For these cases there will be no
+                // entry in validation state to to match and determine the correct key.
+                //
+                // See https://github.com/aspnet/Mvc/issues/7503
+                //
+                // This is to avoid adding validation errors for an 'empty' prefix when a simple
+                // type fails to bind. The fix for #7503 uncovered this issue, and was likely the
+                // original problem being worked around that regressed #7503.
+                string modelName = modelBindingContext.ModelName;
+
+                if (string.IsNullOrEmpty(modelBindingContext.ModelName) &&
+                    (parameter.BindingInfo?.BinderModelName ?? metadata.BinderModelName) == null)
```

Comment:
With @pranavkm's changes in #7564, `parameter.BindingInfo?.BinderModelName` should already reflect `metadata.BinderModelName`.

Reply:
Yes, but I will be first because I believe in myself more.